### DISCUSSION TITLE
Install error : tfel::math::abs redefinition

### DISCUSSION THREAD

**Discussion by arnaud-lejeune** [thelfer/tfel#220] [Q&A] [answered]
HI,

trying to install current version of mfront with "visual 17 2022" , 
I run the .bat scripts form Mfront website to get sources with git and make few changes on compile script compil-vs.bat :
********************************************************************
set TFELHOME=C:\MYDIR\tfel-master\install-vs
rem

mkdir build-vs
cd build-vs
"C:\Program Files\CMake\bin\cmake.exe" ..\tfel -G "Visual Studio 17 2022" -DCMAKE_BUILD_TYPE=Release -Denable-calculix=ON -Denable-ansys=ON -DTFEL_SVN_REVISION=Unknown -DCMAKE_INSTALL_PREFIX=%TFELHOME%
"C:\Program Files\CMake\bin\cmake.exe" --build . --config Release
"C:\Program Files\CMake\bin\cmake.exe" --build . --config Release --target install
pause
*****************************************************************

Install failed. I get the following error :

**************************************
TensorConcept.hxx(126,3): error C2475: 'tfel::math::abs' :  redefinition; 'constexpr' specifier mismatch  [ tfel-master\src\build-vs\src\Math\TFELMath.vcxproj ]
StensorConcept.hxx(90): message : see declaration of  'tfel::math::abs' [ \tfel-master\src\build-vs
\src\Math\TFELMath.vcxproj]
***************************************
I get same error for others functions
tfel::math::trace TensorConcept.hxx(131,3)
tfel::math::computeDeterminantDerivative

and also get error C2995 for template :
TensorConcept.ixx(529,3): error C2995: 'enable_if<0,void>::type tfel::math::polar_decomposition(TensorType &,StensorType &,con
st TensorType2 &)' : function template has already been defined
 (and others and others ...)

I'm an old man not an expert with "new" c++17 norm and compiler. As I assume last version of MFront works on Linux with g++, does anyone suceed in compiling with visual or has a magic option to workaround this error ?

Thank for your help

**Comment by thelfer** [comment, marked as answer]
Hi @arnaud-lejeune,
Visual Studio does not support some constructs that we use in TFEL-4.0. For the moment, please use the rliv-3.4 branch.
Best,
Thomas
